File: terre04.js
// Pair ou impair

const args = process.argv.slice(2)
if (args.length != 1) {
    console.log("Usage: terre04.js <number>")
    process.exit(1)
}

const number = parseInt(args[0])

if (isNaN(number)) {
    console.log(`Expected a valid number got=${arg}`)
    process.exit(1)
}

if (number % 2 === 0) {
    console.log("pair")
} else {
    console.log("impair")
}

File: terre04.exs
# pair ou impair
case System.argv() do
  [arg] ->
    case Integer.parse(arg) do
      {number, _rest} ->
        if rem(number, 2) == 0 do
          IO.puts("pair")
        else
          IO.puts("impair")
          System.exit(1)
        end

      :error ->
        IO.puts("Expected a valid number got=#{arg}")
        System.stop(1)
    end

  _args ->
    IO.puts("Usage: elixir  terre04.js <number>")
    System.stop(1)
end
